packages/lit-dev-content/site/docs/localization/runtime-mode.md

```diff
@@ -0,0 +1,304 @@
+
+class MyElement extends LitElement {
+  constructor() {
+    super();
+    updateWhenLocaleChanges(this);
+  }
+
+  render() {
+    // Whenever setLocale() is called, and templates for that locale have
+    // finished loading, this render() function will be re-invoked.
+    return msg(html`Hello <b>World!</b>`);
+  }
+}
+customElements.define('my-element', MyElement);
+```
+
+{% endswitchable-sample %}
+
+## Status event
+
+The `lit-localize-status` event fires on `window` whenever a locale switch
+starts, finishes, or fails. You can use this event to:
+
+- Re-render when you can't use the `@localized` decorator (e.g. when using the
+  Lit `render` function directly).
+
+- Render as soon as a locale switch begins, even before it finishes loading
+  (e.g. a loading indicator).
+
+- Perform other localization related tasks (e.g. setting a locale preference
+  cookie).
+
+### Event types
+
+The `detail.status` string property tells you what kind of status change has
+occured, and can be either `loading`, `ready`, or `error`:
+
+<dl class="params">
+  <dt class="paramName">loading</dt>
+  <dd class="paramDetails">
+    <p>A new locale has started to load.</p>
+    <p>The <code>detail</code> object contains:</p>
+    <ul>
+      <li><code>loadingLocale: string</code>: Code of the locale that has
+      started loading.</li>
+    </ul>
+    <p>In the case that a second locale is requested before the first one
+    finishes loading, a new <code>loading</code> event is dispatched, and no
+    <code>ready</code> or <code>error</code> event will be dispatched for the
+    first request.</p>
+    <p>A <code>loading</code> status can be followed by a <code>ready</code>,
+    <code>error</code>, or <code>loading</code> status.</p>
+  </dd>
+
+  <dt class="paramName">ready</dt>
+  <dd class="paramDetails">
+    <p>A new locale has successfully loaded and is ready for rendering.</p>
+    <p>The <code>detail</code> object contains:</p>
+    <ul>
+      <li><code>readyLocale: string</code>: Code of the locale that has
+      successfully loaded.</li>
+    </ul>
+    <p>A <code>ready</code> status can be followed only by a
+    <code>loading</code> status.</p>
+  </dd>
+
+  <dt class="paramName">error</dt>
+  <dd class="paramDetails">
+    <p>A new locale failed to load.</p>
+    <p>The <code>detail</code> object contains:</p>
+    <ul>
+      <li><code>errorLocale: string</code>: Code of the locale that failed to
+      load.</li>
+      <li><code>errorMessage: string</code>: Error message from locale load
+      failure.</li>
+    </ul>
+    <p>An <code>error</code> status can be followed only by a
+    <code>loading</code> status.</p>
+  </dd>
+</dl>
+
+### Example of using the status event
+
+```ts
+// Show/hide a progress indicator whenever a new locale is loading,
+// and re-render the application every time a new locale successfully loads.
+window.addEventListener('lit-localize-status', (event) => {
```

Comment:
Is the type of this event automatically available when using localize? I assume it's on the typescript global event interface.